### PR TITLE
 Add new page add_deposit

### LINKED ISSUE #404: [FRONTEND]  Add new page `add_deposit`
🚨 **Attention: Please read the guidelines before applying; otherwise, you will be ignored.** 🚫📋
0. Please, join the telegram group to keep in touch https://t.me/+m0IGRiBvLtsyMzIy 
1. Read guideline [here](https://gist.github.com/djeck1432/f722f5fe518c80b248ff609627da25e5)
2. Link to the Figma - [here](https://www.figma.com/design/WZGStY327tqwLAEhaIXccY/ZkLend-Position)
3. Add `add deposit` subpage in dashboard page
4. Don't add left sidebar, It will be done another contributor
5. Add mobile version of it
6. Send to backend post request `/api/add-extra-deposit/{position_id}` with `position_id` and `amount`, `token_symbol`
7. `Position_id` you can fetch in dashboard page hook
Screenshot 2024-12-18 at 18 02 47

P.S
- Please reuse [Card](https://github.com/djeck1432/spotnet/blob/main/frontend/src/components/Card/Card.jsx) component, that already in use on Dashboard page and adjust its styles according to figma. Make sure both pages still work correctly.

- Please reuse [TokenSelector](https://github.com/djeck1432/spotnet/blob/main/frontend/src/components/TokenSelector.jsx) component that already in use on Form page and adjust its styles according to figma (TokenSelector.jsx should be put in its own folder TokenSelector and tokenSelector.css should be created with isolated styles for that components). Make sure both pages still work correctly.

### PR DESCRIPTION
**Pull request by GoSTEAN**
### Changes made:

- added the add_deposit page

This closes #404

### CODE REVIEW

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -104,6 +105,7 @@ function App() {
             element={walletId ? <Navigate to="/" /> : <Login onConnectWallet={handleConnectWallet} />}
           />
           <Route path="/dashboard" element={<Dashboard telegramId={window?.Telegram?.WebApp?.initData?.user?.id} />} />
+          <Route path="/add_deposit" element={<Add_Deposit />} />
```
Comment (whateverfw) [inline frontend/src/App.jsx:108]: could you please put it under `dashboard/add-deposit` path?

File: frontend/src/pages/spotnet/dashboard/add_deposit.jsx
Comment (whateverfw) [inline frontend/src/pages/spotnet/dashboard/add_deposit.jsx:1]: you should have `add-deposit` folder inside `dashboard`, with `AddDeposit.jsx` and `addDeposit.css`, rename/rework

File: frontend/src/pages/spotnet/dashboard/add_deposit.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import Card from 'components/Card/Card';
+import './add_deposit.css';
+import TokenSelectorDeposit from 'components/TokenSelector/TokenSelector';
+import { ReactComponent as HealthIcon } from 'assets/icons/health.svg';
+import { ReactComponent as EthIcon } from 'assets/icons/ethereum.svg';
+export default function Add_Deposit() {
+   const [amount, setAmount] = useState('0');
+  const handleAmountChange = (e) => {
+    const value = e.target.value;
+    const regex = /^\d*\.?\d*$/;
```
Comment (whateverfw) [inline frontend/src/pages/spotnet/dashboard/add_deposit.jsx:12]: please extract this to a `utils/regex.js` folder and name as a constant with an uppercase with more meaningful name, like `NUMBER_REGEX` or smth

File: frontend/src/pages/spotnet/dashboard/add_deposit.css [hunk 3]
```diff
@@ -0,0 +1,506 @@
+body {
+    font-family:
+      'Rethink Sans',
+      -apple-system,
+      Roboto,
+      Helvetica,
+      sans-serif;
+  }
```
Comment (whateverfw) [inline frontend/src/pages/spotnet/dashboard/add_deposit.css:8]: we already have this in global.css, remove

File: frontend/src/pages/spotnet/dashboard/add_deposit.jsx
Comment (whateverfw) [inline frontend/src/pages/spotnet/dashboard/add_deposit.jsx:1]: please run `yarn format` in order to format your files with prettier

File: frontend/src/pages/spotnet/dashboard/add_deposit.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                  pattern="^\d*\.?\d*$"
+                  className="amount-field"
+                  aria-describedby="currency-symbol"
+                  placeholder="0.00"
+                />
+                <span id="currency-symbol" className="currency">
+                  STRK
+                </span>
+              </div>
+        <button className=" withdraw-button btn">Deposit</button>
```
Comment (whateverfw) [inline frontend/src/pages/spotnet/dashboard/add_deposit.jsx:50]: currently it is not integrated with a backend, as requested in an issues. Please create custom hook `useAddDeposit` and perform actions using `useMutation` hook of `tanstack-query`, as stated in the issue, `position_id` will be returned from `useDashboardData` hook, so take it from there in order to make add deposit request.

File: frontend/src/pages/spotnet/dashboard/add_deposit.jsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  };
+ 
+  return (
+    <div className="deposit-wrapper">
+      <div className="deposit-container">
+        <h1 className="deposit-title1">zkLend Deposit</h1>
+        <div className="main-container-deposit">
+          <div className="top-cards-deposit">
+            <Card label="Health Factor" value="1.47570678" icon={<HealthIcon className="icon" />} />
+            <Card label="Borrow Balance" value="$-55.832665" icon={<EthIcon className="icon" />} />
```
Comment (whateverfw) [inline frontend/src/pages/spotnet/dashboard/add_deposit.jsx:25]: cards do not look like in figma, everything is different, pleae make changes to css and verify it looks just like in figma on both mobile and desktop

File: frontend/src/components/TokenSelector/TokenSelector.jsx
Comment (whateverfw) [inline frontend/src/components/TokenSelector/TokenSelector.jsx:1]: that's good you've created sepearate folder for token selector, but please make styles according to figma, also remove duplicated `components/TokenSelector.jsx` and use this new one in `Form.jsx` as well as on deposit page (that you already do)

File: frontend/src/components/TokenSelector/TokenSelector.css
Comment (whateverfw) [inline frontend/src/components/TokenSelector/TokenSelector.css:1]: please name your classess correctly, most likely with `token` prefix, it should not have anything related to form in the name. Also css completely do not match figma on mobile/desktop. Please fix

File: frontend/src/pages/spotnet/dashboard/add_deposit.jsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+                  onChange={handleAmountChange}
+                  pattern="^\d*\.?\d*$"
+                  className="amount-field"
+                  aria-describedby="currency-symbol"
+                  placeholder="0.00"
+                />
+                <span id="currency-symbol" className="currency">
+                  STRK
+                </span>
+              </div>
```
Comment (whateverfw) [inline frontend/src/pages/spotnet/dashboard/add_deposit.jsx:49]: please fix styles for this input, currently it do not match figma, font is too small and

File: frontend/src/pages/spotnet/dashboard/add_deposit.css [hunk 7]
```diff
@@ -0,0 +1,506 @@
...
+  @media (max-width: 680px){  
+    .deposit-container {
+  
+      /* margin-top: -120px;
+      margin-left: 0px; */
+    }
+  
+    /* .deposital-title {
+      font-size: 24px;
+    } */
```
Comment (whateverfw) [inline frontend/src/pages/spotnet/dashboard/add_deposit.css:261]: do not keep commented code

**Review by whateverfw — CHANGES_REQUESTED** [review]

File: frontend/src/components/TokenSelector.jsx
Comment (whateverfw) [inline frontend/src/components/TokenSelector.jsx:1]: you should work only with single `TokenSelector`, please remove this one and work/use only your newly created TokenSelector.jsx

**Review by whateverfw — CHANGES_REQUESTED** [review]
How is the progress with a tasks?

File: frontend/src/components/TokenSelector.jsx
Comment (GoSTEAN) [reply frontend/src/components/TokenSelector.jsx:1]: Okay, working on it 
Will be done soon

**Comment by djeck1432** [comment]
@GoSTEAN Any updates?
all PR should be merged today, if you not finish your task you will be rewarded partially

**Comment by whateverfw** [comment]
@GoSTEAN sync with main and fix merge conflicts

**Comment by GoSTEAN** [comment]
Presently fixing the conflicts , Will be pushing soon

**Comment by GoSTEAN** [comment]
@whateverfw I have resolved the conflict on my local machine but it's still reflecting on Github. Pls what should i do ?

**Comment by whateverfw** [comment]
@GoSTEAN I'm not sure, did you resolve it correctly and pushed the changes to git?

**Comment by djeck1432** [comment]
@GoSTEAN To save time, open new open, but do NOT close this one

**Comment by GoSTEAN** [comment]
@djeck1432 I have created a new pull request

**Comment by djeck1432** [comment]
Close this PR as a duplicated

**Pull request closed without merging**